src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/StartedNode.kt

```diff
@@ -68,9 +70,20 @@ class StartedNode(
             resultsTransport = this.resultsTransport,
             unpackedProduct = this.unpackedProduct,
             monitoringProcesses = this.monitoringProcesses,
-            ssh = this.ssh
+            ssh = this.ssh,
+            jdk = this.jdk
         )
     }
 
+    fun stopNode() {
+        ssh.newConnection()
+            .use { ssh ->
+                ssh.safeExecute(
+                    "${jdk.use()}; ./${unpackedProduct}/bin/stop-jira.sh",
```

Comment:
Oh snap, `jdk` is needed for this? I thought it's all bash scripts.
Unfortunately, adding this to the constructor breaks old constructor calls. Sure, from `ijpt` perspective we only see it called from awsinfra, so in theory it wouldn't break any caller. But it's a nasty surprise for anybody in another repo or another branch. And precedent for a dangerous shortcut.

First, I'd see if it can just skip the `jdk.
If not, then let's hack the impl instead of hacking the API:
1. keep the new primary constructor
2. restore the old constructor as a secondary constructor, deprecate it
3. use a "default JDK" in the old constructor, even lazy-install it if it's necessary

Reply:
I already tried without `jdk` and got this message: "No JRE_HOME or JAVA_HOME environment variable is set - attempting to just run java command" but I need to verify if it failed or maybe somehow succeeded.

Reply:
We can do optional `jdk`, i.e. nullable internally, and throw / return "no success" when we don't have it. It's a new feature that is supported only if you use the new builder constructor.

Reply:
Ok, I have checked how it works without JDK and even though no explicit error is thrown, the script doesn't work. After execution, Jira is still running.

Reply:
Noice :muscle: 
Also, we're supposed to simulate a Jira admin. I doubt they write `JAVA_HOME=dfhsdkfhsdkf ./bin/start-jira.sh` every time. I bet they configure it once in `~/.bashrc` during JDK/Jira installation.

Reply:
Yup, we probably should register the env variable for every process as part of Java installation